.travis.yml

```diff
@@ -42,25 +43,46 @@ matrix:
     - php: 7.3
       env:
         - DEPS=latest
-
-services: mongodb
+    - php: 7.4
+      env:
+        - DEPS=lowest
+    - php: 7.4
+      env:
+        - DEPS=locked
+    - php: 7.4
+      env:
+        - DEPS=latest
 
 before_install:
   - travis_retry composer self-update
+  - shopt -s expand_aliases
   - if [[ $DEPS == 'lowest' ]]; then travis_retry composer remove $COMPOSER_ARGS --no-update --dev $LOWEST_DEPS_REMOVE ; fi
+  - alias composer="travis_retry docker-compose run --rm php composer"
+  - alias php="travis_retry docker-compose run -e TRAVIS -e TRAVIS_JOB_ID --rm php php"
+  - docker-compose build --build-arg PHP_VERSION=${TRAVIS_PHP_VERSION} --build-arg XDEBUG=${TEST_COVERAGE:+1} --no-cache php
+  - composer --version
+  - php -v
+  - php -m
 
 install:
-  - yes '' | pecl -q install -f mongodb
+  - composer install $COMPOSER_ARGS --ignore-platform-reqs
+  - if [[ $DEPS == 'latest' ]]; then travis_retry composer config "platform.ext-mongo" "1.6.16" && composer update $COMPOSER_ARGS ; fi
+  - if [[ $DEPS == 'lowest' ]]; then travis_retry composer config "platform.ext-mongo" "1.6.16" && composer update --prefer-lowest --prefer-stable $COMPOSER_ARGS ; fi
+
   - travis_retry composer config "platform.ext-mongo" "1.6.16" && composer require --dev $COMPOSER_ARGS $ADAPTER_DEPS
```

Comment:
:thinking: line 68 uses an alias, right? this one does since `composer` is not the first word. I gave a cursory glance at the length discussion, and I don't think this was intended when I read this comment: https://github.com/zfcampus/zf-apigility-doctrine/pull/308#issuecomment-383847819

Reply:
Correction, the bare metal composer is in use for `composer config "platform.ext-mongo" "1.6.16"`, the docker one is in use for `composer require --dev $COMPOSER_ARGS $ADAPTER_DEPS`

Reply:
The part before the && seems unecessary, the `ext-mongo` version is already specified in `composer.json` since a0179078e812b446c72f4c95559a04c007483a67

Reply:
As for the && that's not just attrition.  I had repeated errors when that was not included with each docker command.

Are you saying the platform.ext-mongo is being ran locally instead of in docker?

Reply:
yes, because aliases are not resolved inside of a command:
```
╭─~ 
╰$ alias foo=bar   
╭─~ 
╰$ foo               
zsh: bar: command not found...
^C
╭─~ 
╰$ echo foo    
foo
╭─~ 
╰$ echo foo && foo
foo
zsh: bar: command not found...

```